app/query-client.ts

```diff
@@ -21,3 +22,10 @@ export function getQueryClient() {
   }
   return browserQueryClient;
 }
+
+export function getServerQueryClient() {
```

Comment:
If we want to keep the singleton for this (I probably wouldn't for now), I would rename `browserQueryClient` to `singletonQueryClient` (or `queryClientInstance`) and add optional bool param (default false) to getQueryClient called `singleton` then I would just change `if (isServer) {` to `if (isServer && !singleton) {` and delete `getServerQueryClient`